daemon/containerd/image_pull.go

```diff
@@ -70,6 +91,13 @@ func (i *ImageService) PullImage(ctx context.Context, image, tagOrDigest string,
 	infoHandler := snapshotters.AppendInfoHandlerWrapper(ref.String())
 	opts = append(opts, containerd.WithImageHandlerWrapper(infoHandler))
 
+	progress.Message(out, tagOrDigest, "Pulling from "+reference.Path(ref))
+
+	var oldDigest digest.Digest
+	if err == nil {
```

Comment:
I think this was supposed to refer to err from `old, err := i.resolveDescriptor(ctx, ref.String())`

but it actually uses the err from 
```go
available, _, _, _, err := images.Check(ctx, i.client.ContentStore(), old, p)
if err != nil {
     return err
}
```

which will always be nil.

Reply:
Anyway, the usage of `err` defined 30 lines will probably be broken in the next PR which adds some code in between. We need some boolean or nil check here.